File: memo-service/app/models/Memo.scala
package models

import java.time.LocalDateTime
import java.util.UUID

import reactivemongo.bson.{BSONDocument, BSONDocumentWriter}

case class MemoForm(title: String, content: String, summary: String, tags: Set[String])
case class Memo(
                 id: UUID = UUID.randomUUID,
                 title: String = "",
                 content: String = "",
                 summary: String = "",
                 tags: Set[String] = Set.empty,
                 created_time: LocalDateTime = LocalDateTime.now(),
                 updated_time: LocalDateTime = LocalDateTime.now()
               )

object MemoForm {
  implicit val memoWriter: BSONDocumentWriter[MemoForm] =
    BSONDocumentWriter[MemoForm] { memoForm: MemoForm =>
      BSONDocument(
        "title" -> memoForm.title,
        "content" -> memoForm.content,
        "summary" -> memoForm.summary,
        "tags" -> memoForm.tags
      )
    }
}

File: memo-service/app/models/memo.js
/*!
 * Module dependencies
 */

const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const uuid = require('uuid4');

/**
 * User schema
 */

const MemoSchema = new Schema({
  uuid: { type: String, default: uuid() },
  title: { type: String, default: '' },
  content: { type: String, default: '' },
  summary: { type: String, default: '' },
  tags: { type: [String], default: '' },
  user_id: { type: String },
  created_time: { type: Date },
  updated_time: { type: Date }
});

/**
 * Add your
 * - pre-save hooks
 * - validations
 * - virtuals
 */

/**
 * Methods
 */

MemoSchema.method({});

/**
 * Statics
 */

MemoSchema.static({});

/**
 * Register
 */

mongoose.model('Memo', MemoSchema);
